qdrant_client/local/payload_value_extractor.py

```diff
@@ -63,6 +63,53 @@ def _get_value(data: Any, k_list: List[str]) -> None:
     return result if result else None
 
 
+def set_value_by_key(payload: Dict[str, Any], new_value: Dict[str, Any], key: str) -> None:
+    keys = key.split(".")
+
+    def _set_value(data: Any, k_list: List[str], new_value: Dict[str, Any]) -> None:
+        if not k_list:
+            return
+
+        k = k_list.pop(0)
+
+        if k.endswith("]"):
+            k, index = k.split("[")
+            if k not in data:
+                data[k] = []
+                return
+
+            data = data.get(k)
+
+            if not isinstance(data, list):
+                return
+
+            index = index.strip("]")
+            if index == "":
+                for item in data:
+                    if isinstance(item, dict) and len(k_list) == 0:
+                        item.update(new_value)
+                    else:
+                        _set_value(item, k_list.copy(), new_value)
```

Comment:
This fragment appears three times in this function with slight changes.
Can it be refactored to appear only once? Perhaps it could be moved to the top of this function.